src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs

```diff
@@ -152,5 +152,134 @@ public static void GetHRForException()
                 Marshal.GetExceptionForHR(e.HResult);
             }
         }
+
+        [Fact]
+        public static void GenerateGuidForType()
+        {
+             Assert.Throws<PlatformNotSupportedException>(() => Marshal.GenerateGuidForType(typeof(System.Int32)));        
```

Comment:
I don't see a strong use case for this API - as a matter of fact, I've never see people calling this API explicitly. In general people should use type.Guid to retrieve the GUID (that are either generated by the runtime, or specified in GUIDAttribute) rather than generating the GUID. Also, we don't have good AOT support for dynamic GUID generation, so I don't want to add more debt here. 
@stephentoub Are you aware of interesting use cases for this API?